moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -277,8 +278,11 @@ bool planning_pipeline::PlanningPipeline::generatePlan(const planning_scene::Pla
       m.action = visualization_msgs::Marker::DELETEALL;
       arr.markers.push_back(m);
 
+      auto path_constraints_resolved = req.path_constraints;
+      kinematic_constraints::resolveConstraintFrames(res.trajectory_->getFirstWayPoint(), path_constraints_resolved);
```

Comment:
We end up resolving the constraints 2 times, for this method and also in the `adaptAndPlan` (if the plugin is in the chain). Would it not be better to create a `PlanningRequestAdapter` for this [section of code](https://github.com/moveit/moveit/blob/81822f2cd660739c10e11b6bee9e19448da0dd28/moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp#L270-L354) ? This would remove the need to resolve the constraint frames an additional time, but I don't know the implication of doing so.

Reply:
The problem is that the planning request adapters don't affect the original request in this code, leaving the frames unresolved.

Reply:
In #2975, I was proposing exactly that. But thinking about it now, one would need to copy the hole `MotionPlanRequest` each time before passing it with the modified frames. With your solution there is no request copy, the frames are only evaluated 2 times, which seems to have less overhead.

I was talking about setting the `isPathValid` and `isStateValid` at the end of the chain, that way the given frames would have been already resolved and accessible. But these checks seems to be coupled with the planning pipeline ([contact_publiser_](https://github.com/captain-yoshi/moveit/blob/fix-resolved-constraint-frames/moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp#L338)).
```xml
  <arg name="planning_adapters"
       default="
                default_planner_request_adapters/isStateValid
                default_planner_request_adapters/isPathValid
                default_planner_request_adapters/AddTimeParameterization
                default_planner_request_adapters/ResolveConstraintFrames
                default_planner_request_adapters/FixWorkspaceBounds
                default_planner_request_adapters/FixStartStateBounds
                default_planner_request_adapters/FixStartStateCollision
                default_planner_request_adapters/FixStartStatePathConstraints"
                />

```

Reply:
I think, your PR #2975 is the easiest way to go. Sorry for not responding to this yet.
MoveIt2 implemented path validation as a separate adapter - a very clean solution.

Reply:
I factored this fixup out and leave that to #2975.